File: typescript/src/info.ts
/** POST /v1/info — free metadata lookup (does not consume quota). */

import { VideoFetch } from "./client";
import { VideoInfo } from "./types";

export class InfoResource {
  constructor(private client: VideoFetch) {}

  async lookup(url: string): Promise<VideoInfo> {
    return this.client.request<VideoInfo>("POST", "/v1/info", { json: { url } });
  }
}

File: go/info.go
package videofetch

import "context"

// InfoService provides free metadata lookups. Access via client.Info.
type InfoService struct{ client *Client }

// Lookup resolves video metadata (title, duration, available formats) without
// consuming quota. Accepts a full URL or bare video ID.
func (s *InfoService) Lookup(ctx context.Context, url string) (*VideoInfo, error) {
	var info VideoInfo
	if err := s.client.request(ctx, "POST", "/v1/info", map[string]string{"url": url}, &info); err != nil {
		return nil, err
	}
	return &info, nil
}
